src/dashboard/post_distribute/post_distribute.py

```diff
@@ -0,0 +1,51 @@
+"""Handler for validating the payload contents and submitting to queue"""
+
+import json
+import os
+
+import boto3
+import requests
+
+from shared import functions
+
+sns_client = boto3.client("sns", os.environ.get("AWS_REGION"))
+valid_chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~:/?#[]@!$&'()*+,"
+
+
+def validate_github_url(config):
+    if not config["url"].startswith("https://github.com/smart-on-fhir/") or any(
+        c not in valid_chars for c in config["url"]
+    ):
+        raise ValueError(f"{config['url']} is not an official Cumulus study.")
+    res = requests.get(config["url"], timeout=10)
+    if res.status_code != 200:
+        raise ValueError(f"{config['url']} is not a valid git repository")
+    if "tag" in config and config["tag"] is not None:
```

Comment:
If tag is not provided, this method just silently does nothing - maybe throw a ValueError or similar?

Also - do you really want `is None` here and allow an empty string tag? Maybe just `if config.get("tag");` is sufficient.

Reply:
this was intentional - if you don't provide a tag, you want the tip of main. can be less kind if you think it's error prone.

Reply:
Oh I guess this was doing more than just pinging URLs and actually doing some downloads. I think I just misread it.

Reply:
Yeah that's fine, it goes back to my confusion earlier - where like this get request _did_ something. I see now what's up. :+1: